### ISSUE TITLE
Projected LFRic cubesphere C12 produces invalid arcsin error

### ISSUE THREAD

**Issue by trexfeathers** [bjlittle/geovista#66]
## 🐛 Bug Report

Plotting a C12 LFRic cubesphere on a planar projection produces the error:

```
...geovista/common.py:299: RuntimeWarning: invalid value encountered in arcsin
  lats = np.degrees(np.arcsin(xyz[:, 2] / radius))
```

This is accompanied by an invalid looking plot with Mollweide, although not with Plate Carree.

I have not seen this behaviour with C72 cubespheres. Might be worth some experiments at other resolutions.

## How To Reproduce

<details>
<summary>Expand for code</summary>

```python
import geovista
from geovista import Transform
from geovista.geoplotter import GeoPlotter
from iris import load_cube
from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD
from iris.tests import get_data_path

print(geovista.__version__)

file_path = get_data_path(
    [
        "NetCDF",
        "unstructured_grid",
        "lfric_ngvat_2D_1t_face_half_levels_main_conv_rain.nc",
    ]
)

with PARSE_UGRID_ON_LOAD.context():
    my_cube = load_cube(file_path, "conv_rain")

face_node = my_cube.mesh.face_node_connectivity
lons, lats = my_cube.mesh.node_coords
indices = face_node.indices_by_location()
my_polydata = Transform.from_unstructured(
    lons.points, lats.points, indices, start_index=face_node.start_index,
)

for proj in ("moll", "eqc"):
    my_plotter = GeoPlotter(crs=f"+proj={proj}")
    my_plotter.add_mesh(my_polydata, show_edges=True)
    my_plotter.camera_position = "xy"
    my_plotter.show(screenshot=f"{proj}.png")

```

</details>

## Expected behaviour

No error raised, and valid plots produced, given this will become a widely used format.

## Screenshots

<details>
<summary>Expand for screenshots</summary>

![moll](https://user-images.githubusercontent.com/40734014/163418730-8f2a6fb8-66dd-4347-a08d-c16a006c6da1.png)
![eqc](https://user-images.githubusercontent.com/40734014/163418726-b379b3e3-86d0-4705-adf7-2938c1a2b4a3.png)

</details>

## Environment 

* OS & Version - Linux RHEL7
* GeoVista Version - `0.1a1.dev313`

**Comment by bjlittle** [comment]
This is a rounding issue, which I've seen before 👍

**Comment by bjlittle** [comment]
@trexfeathers This issue has been resolved by b415e50cc5fc84a0e6d28004727c521978158011.

Additionally, note that pole cell management is more stable (and correct). This is more obvious in the Equidistant Cylindrical (Plate Carrée) projection for your C12 cube-sphere example (see below). 

Thanks again for providing the code to replicate this issue, it really helped :100: :1st_place_medal: 

![moll](https://user-images.githubusercontent.com/2051656/215231913-e85f8e15-a47b-41cd-abbd-bf703f0b84b1.png)

![eqc](https://user-images.githubusercontent.com/2051656/215231921-98d9d559-b543-447c-a27e-24dcbfa521e1.png)

I'm going to close this issue, but please re-open it if you don't believe it's resolved.

:beers:

**Comment by bjlittle** [comment]
@all-contributors please add @trexfeathers for question
